server/middleware/github-griller/github.utils.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+var Promise = require('bluebird');
+var Board = require('mongoose').model('Board');
+var mongoose = require('mongoose')
+
+// FIXME This function is just an outline of what it could be
+exports.dataUpsert = function(dataArr, model) {
+  Promise.map(dataArr, function(dataItem, index) {
+        if (!dataItem) dataItem = { githubId: null }
+
+        return model.findOneAndUpdate({
+          githubId: dataItem.githubId
+        }, data[index], {
+          upsert: true,
+          new: true
+        })
+      })
+      .then(function(dataUpserted) {
+        res.send(dataUpserted)
+      })
+};
+
+// A generic function for getting the remaining pages of a github request. Expects a github client, a config object, and a function to get data from github
+exports.getRemainingPages = function(gitRes, concatData=[]) {
```

Comment:
@ssarda77 This is where I've put the getRemainingPages function.  I figured it would be good to have a 'utils' object that has a few different functions that are just used by other parts of the GithubGriller module.